framework/include/kokkos/functions/KokkosPiecewiseConstant.h

```diff
@@ -0,0 +1,103 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "KokkosPiecewiseTabularBase.h"
+#include "KokkosUtils.h"
+
+using Moose::Kokkos::Real3;
+
+/**
+ * Function which provides a piecewise constant interpolation of a provided (x,y) point data set.
+ */
+class KokkosPiecewiseConstant : public KokkosPiecewiseTabularBase
+{
+public:
+  static InputParameters validParams();
+
+  KokkosPiecewiseConstant(const InputParameters & parameters);
+
+  KOKKOS_FUNCTION Real value(Real t, Real3 p) const;
+  KOKKOS_FUNCTION Real integral() const;
+  KOKKOS_FUNCTION Real average() const;
+
+private:
+  /// Enum for which direction to apply values
+  const enum class Direction { LEFT, RIGHT, LEFT_INCLUSIVE, RIGHT_INCLUSIVE } _direction;
+};
+
+KOKKOS_FUNCTION inline Real
+KokkosPiecewiseConstant::value(Real t, Real3 p) const
+{
+  using Moose::Kokkos::Utils::sign;
+
+  const Real x = _has_axis ? p(_axis) : t;
+
+  const auto len = functionSize();
+  const Real tolerance = 1.0e-14;
```

Comment:
```suggestion
  constexpr Real tolerance = 1.0e-14;
```
My understanding is that all the device compilers we care about support this